.github/workflows/ci.yml

```diff
@@ -0,0 +1,84 @@
+name: CI
+on: [pull_request, push]
+jobs:
+  test:
+    name: Test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        toolchain: [stable, nightly]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Rust
+        run: |
+          rustup toolchain add ${{ matrix.toolchain }} --component llvm-tools-preview
+          rustup override set ${{ matrix.toolchain }}
+      - name: Configure cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: test-${{ runner.os }}-cargo-${{ matrix.toolchain }}-${{ hashFiles('**/Cargo.lock') }}
+      - name: Test
+        run: cargo test --lib
+  lint:
```

Comment:
There is some redundancy between `lint` and `pre-commit`. `pre-commit` is running rustfmt and clippy too.
Let's remove the `lint` entry and just use `pre-commit` for linting (but rename the `pre-commit` entry to `lint`, since the fact we're using `pre-commit` for linting is just an "implementation detail" which might change).

Reply:
There definitely is. I'll remove the lint action and rename the pre-commit job to lint.

Originally I added the lint as I copied the config from one of my own config and then adjusted it.